internal/datastore/crdb/stats.go

```diff
@@ -76,37 +53,84 @@ func (cds *crdbDatastore) Statistics(ctx context.Context) (datastore.Stats, erro
 		return datastore.Stats{}, err
 	}
 
-	// NOTE: this is a stop-gap solution to prevent panics in telemetry collection
-	if relCount < 0 {
-		relCount = 0
-	}
-
-	return datastore.Stats{
-		UniqueID:                   uniqueID,
-		EstimatedRelationshipCount: uint64(relCount),
-		ObjectTypeStatistics:       datastore.ComputeObjectTypeStats(nsDefs),
-	}, nil
-}
+	if cds.analyzeBeforeStatistics {
+		if err := cds.readPool.BeginTxFunc(ctx, pgx.TxOptions{AccessMode: pgx.ReadOnly}, func(tx pgx.Tx) error {
+			if _, err := tx.Exec(ctx, "ANALYZE "+tableTuple); err != nil {
+				return fmt.Errorf("unable to analyze tuple table: %w", err)
+			}
 
-func updateCounter(ctx context.Context, tx pgx.Tx, change int64) (datastore.Revision, error) {
-	counterID := make([]byte, 2)
-	// nolint:gosec
-	// G404 use of non cryptographically secure random number generator is not concern here,
-	// as this is only used to randomly distributed the counters across multiple rows and reduce write row contention
-	_, err := rand.New(rng).Read(counterID)
-	if err != nil {
-		return datastore.NoRevision, fmt.Errorf("unable to select random counter: %w", err)
+			return nil
+		}); err != nil {
+			return datastore.Stats{}, err
+		}
 	}
 
-	sql, args, err := upsertCounterQuery.Values(counterID, change).ToSql()
-	if err != nil {
-		return datastore.NoRevision, fmt.Errorf("unable to prepare upsert counter sql: %w", err)
-	}
+	var estimatedRelCount uint64
+	if err := cds.readPool.QueryFunc(ctx, func(ctx context.Context, rows pgx.Rows) error {
+		hasRows := false
+
+		for rows.Next() {
+			hasRows = true
+			values, err := rows.Values()
+			if err != nil {
+				log.Warn().Err(err).Msg("unable to read statistics")
+				return nil
+			}
+
+			// Ensure this the correct row.
```

Comment:
Can you expand on this? What other rows come if you are asking for stats of the relation tuple table?

Reply:
Updated the var name and comment